.eslintrc.json

```diff
@@ -0,0 +1,155 @@
+{
+	"root": true,
+	"env": {
+		"browser": true,
+		"es6": true,
+		"node": true
+	},
+	"extends": [
+		// TODO: consider extending the recommended by ts rules and overriding them where necessary
+		// 'eslint:recommended',
+		// 'plugin:@typescript-eslint/recommended'
```

Comment:
After this PR is merged, I will log an issue to dig deeper into eslint as this was not the idea of the PR, it just happened because tslint was throwing after I upgraded ts and the packages around it.